packages/api-db-mongodb/src/db/payment.ts

```diff
@@ -0,0 +1,204 @@
+import {
+  ConnectionResult,
+  InputCursorType,
+  LimitType,
+  SortOrder,
+  CreatePaymentArgs,
+  DBPaymentAdapter,
+  DeletePaymentArgs,
+  GetPaymentsArgs,
+  OptionalPayment,
+  Payment,
+  PaymentSort,
+  UpdatePaymentArgs
+} from '@wepublish/api'
+
+import {Collection, Db, FilterQuery, MongoCountPreferences} from 'mongodb'
+
+import {CollectionName, DBPayment} from './schema'
+import {Cursor} from './cursor'
+import {MaxResultsPerPage} from './defaults'
+
+export class MongoDBPaymentAdapter implements DBPaymentAdapter {
+  private payment: Collection<DBPayment>
+  private locale: string
+
+  constructor(db: Db, locale: string) {
+    this.payment = db.collection(CollectionName.Payments)
+    this.locale = locale
+  }
+
+  async createPayment({input}: CreatePaymentArgs): Promise<Payment> {
+    const {ops} = await this.payment.insertOne({
+      createdAt: new Date(),
+      modifiedAt: new Date(),
+      intentID: input.intentID,
```

Comment:
What does intent refers to?

Reply:
Stripe used that terminology. They call it paymentIntent. A thing that wants to make a payment.

Intent => intention or purpose.